app/kubernetes/react/views/networks/ingresses/CreateIngressView/CreateIngressView.tsx

```diff
@@ -0,0 +1,588 @@
+        ingressesResults.data || [],
+        ruleCounterByNamespace,
+        hostWithTLS,
+      ];
+    }, [ingressesResults.data, namespace]);
+
+  const namespacesOptions: Option<string>[] = [
+    { label: 'Select a namespace', value: '' },
+  ];
+  Object.entries(namespacesResults?.data || {}).forEach(([ns, val]) => {
+    if (!val.IsSystem) {
+      namespacesOptions.push({
+        label: ns,
+        value: ns,
+      });
+    }
+  });
+
+  const clusterIpServices = servicesResults.data?.filter(
+    (s) => s.Type === 'ClusterIP'
+  );
+  const servicesOptions = clusterIpServices?.map((service) => ({
+    label: service.Name,
+    value: service.Name,
+  }));
+
+  const serviceOptions = [
+    { label: 'Select a service', value: '' },
+    ...(servicesOptions || []),
+  ];
+  const servicePorts = clusterIpServices
+    ? Object.fromEntries(
+        clusterIpServices?.map((service) => [
+          service.Name,
+          service.Ports.map((port) => ({
+            label: String(port.Port),
+            value: String(port.Port),
+          })),
+        ])
+      )
+    : {};
+
+  const existingIngressClass = ingressControllersResults.data?.find(
+    (i) => i.ClassName === ingressRule.IngressClassName
+  );
+  const ingressClassOptions: Option<string>[] = [
+    { label: 'Select an ingress class', value: '' },
+    ...(ingressControllersResults.data?.map((cls) => ({
+      label: cls.ClassName,
+      value: cls.ClassName,
+    })) || []),
+  ];
+  if (!existingIngressClass && ingressRule.IngressClassName) {
+    ingressClassOptions.push({
+      label: !ingressRule.IngressType
+        ? `${ingressRule.IngressClassName} - NOT FOUND`
+        : `${ingressRule.IngressClassName} - DISALLOWED`,
+      value: ingressRule.IngressClassName,
+    });
+  }
+
+  const matchedConfigs = configResults?.data?.filter(
+    (config) =>
+      config.SecretType === 'kubernetes.io/tls' &&
+      config.Namespace === namespace
+  );
+  const tlsOptions: Option<string>[] = [
+    { label: 'No TLS', value: '' },
+    ...(matchedConfigs?.map((config) => ({
+      label: config.Name,
+      value: config.Name,
+    })) || []),
+  ];
+
+  useEffect(() => {
+    if (!!params.name && ingressesResults.data && !ingressRule.IngressName) {
+      // if it is an edit screen, prepare the rule from the ingress
+      const ing = ingressesResults.data?.find(
+        (ing) => ing.Name === params.name
+      );
+      if (ing) {
+        const type = ingressControllersResults.data?.find(
+          (c) => c.ClassName === ing.ClassName
+        )?.Type;
+        const r = prepareRuleFromIngress(ing);
+        r.IngressType = type;
+        setIngressRule(r);
+      }
+    }
+  }, [
+    params.name,
+    ingressesResults.data,
+    ingressControllersResults.data,
+    ingressRule.IngressName,
+  ]);
+
+  useEffect(() => {
+    if (namespace.length > 0) {
+      validate(ingressRule, ingressNames || []);
+    }
+  }, [ingressRule, namespace, ingressNames]);
```

Comment:
Is it possible to fix this eslint warning?

Reply:
it is asking for validate method and reference of a method changes on every render so it it not good idea to pass just a method. I can either `useCallback` for it but I guess it is not a big deal.